### PR TITLE
Make sure smoke test works in multicluster

### PR DESCRIPTION
**Pull request by pu239ppy**
The block of code collecting runtime information needs to run at compile time

### CODE REVIEW

File: cookbooks/bcpc-hadoop/recipes/smoke_test_wrapper.rb [hunk 1]
```diff
@@ -27,6 +27,28 @@
...
+fs = "hdfs://#{chef_env}"
+rm = if resource_managers.length > 1 then chef_env else resource_managers[0] end
+thrift_uris = node[:bcpc][:hadoop][:hive_hosts]
+  .map { |s| 'thrift://' + float_host(s[:hostname]) + ':9083' }.join(",")
+node.default['hadoop_smoke_tests']['carbon-line-receiver'] = node[:bcpc][:graphite][:ip]
+node.default['hadoop_smoke_tests']['carbon-line-port'] = node[:bcpc][:graphite][:relay_port]
+node.default['hadoop_smoke_tests']['oozie_hosts'] = node[:bcpc][:hadoop][:oozie_hosts].map do | entry | float_host(entry['hostname']) end
+node.default['hadoop_smoke_tests']['wf_path'] = "hdfs://Test-Laptop/user/#{test_user}/oozie-smoke-tests/wf"
+node.default['hadoop_smoke_tests']['wf']['co_path'] = "hdfs://Test-Laptop/user/#{test_user}/oozie-smoke-tests/co"
+node.default['hadoop_smoke_tests']['wf']['rm'] = rm
```
Comment (vt0r) [inline cookbooks/bcpc-hadoop/recipes/smoke_test_wrapper.rb]: Should these strings be quoted?

**Comment by cbaenziger** [comment]
@pu239ppy Why do we need to collect this info earlier than later? (Also, I thought I've seen working smoke tests post multicluster?)

**Comment by aespinosa** [comment]
variables in https://github.com/bloomberg/chef-bach/blob/master/cookbooks/smoke-tests/recipes/oozie_smoke_test.rb#L22-L26 seems to be referred at compile time.

**Comment by cbaenziger** [comment]
Thanks for the motivation on this @aespinosa

File: cookbooks/bcpc-hadoop/recipes/smoke_test_wrapper.rb [hunk 2]
```diff
@@ -27,6 +27,28 @@
 
 include_recipe "bcpc-hadoop::smoke_test_user"
 
+chef_env = node.environment
+resource_managers = node[:bcpc][:hadoop][:rm_hosts].map do |rms| float_host(rms.hostname) end
+zookeeper_quorum = node[:bcpc][:hadoop][:zookeeper][:servers].map do |zks| float_host(zks.hostname) end
+fs = "hdfs://#{chef_env}"
+rm = if resource_managers.length > 1 then chef_env else resource_managers[0] end
+thrift_uris = node[:bcpc][:hadoop][:hive_hosts]
+  .map { |s| 'thrift://' + float_host(s[:hostname]) + ':9083' }.join(",")
```
[the remark this answers is not in the document]
Comment (pu239ppy) [reply cookbooks/bcpc-hadoop/recipes/smoke_test_wrapper.rb]: You've proven yourself wrong here 
https://github.com/bloomberg/chef-bach/pull/817

**Comment by pu239ppy** [comment]
@aespinosa I am not sure I see what problem this would cause.

File: cookbooks/bcpc-hadoop/recipes/smoke_test_wrapper.rb (hunk 1 above)
Comment (pu239ppy) [reply cookbooks/bcpc-hadoop/recipes/smoke_test_wrapper.rb]: They are variables

**Comment by pu239ppy** [comment]
Rebased on master

File: cookbooks/bcpc-hadoop/recipes/smoke_test_wrapper.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (vt0r) [reply cookbooks/bcpc-hadoop/recipes/smoke_test_wrapper.rb]: Lol whoops I apparently can't read. Just kidding

File: cookbooks/bcpc-hadoop/recipes/smoke_test_wrapper.rb (hunk 1 above)
Comment (vt0r) [reply cookbooks/bcpc-hadoop/recipes/smoke_test_wrapper.rb]: Another reading comprehension failure on my part. Apologies.

**Pull request merged**